grommet/grommet
# LoginForm: Disable Login button when onSubmit is undefined

Signed-off-by: Mike Turley <mturley@hpe.com>



#### What does this PR do?

Removes the type="submit" attribute from the Button on the LoginForm when the onSubmit prop is undefined, to ensure the button will not be enabled unless an onSubmit function is provided.  This fixes https://github.com/grommet/grommet/issues/1295, which has been a problem since the commit https://github.com/grommet/grommet/commit/f1b2a715b314f82c275fac92bafbdbb109863761 made all buttons with type="submit" or "reset" enabled regardless of their onClick props.

#### Where should the reviewer start?

LoginForm component

#### What testing has been done on this PR?

My own test project, and also added a new unit test and verified that existing snapshot tests are not broken by this change.

#### How should this be manually tested?

Create a LoginForm with no onSubmit prop, and verify that the login button is disabled.

#### Any background context you want to provide?

This is important for providing visual feedback that the form is in the process of being submitted.  Without this fix, the form looks frozen/broken during login.

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/1295

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?

It couldn't hurt, but not necessarily.

#### Should this PR be mentioned in the release notes?

Probably.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible
